### DISCUSSION TITLE
Secrets manager

### DISCUSSION THREAD

**Discussion by modem7** [dani-garcia/vaultwarden#3368] [General]
Heya,

Will vaultwarden eventually be compatible with the secrets manager that's been announced recently?

Thanks!

**Comment by BlackDex** [comment]
No. It's not open source licensed, and not build for the open source web-vault.

**Comment by mon5termatt** [comment]
https://bitwarden.com/products/secrets-manager/

![image](https://user-images.githubusercontent.com/43628254/227754898-f89b2e56-137c-425c-93d2-d72d351d6dab.png)

might have to run it separate but looks like they will make it hostable

**Reply by BlackDex** [reply]
Self hosting is not the same as Open Source.
The Bitwarden self hosted instance still needs a license for Enterprise features.

**Reply by etfz** [reply]
I don't know whether that page was edited, but it clearly says it is open source, and that the source is available on GitHub.
https://bitwarden.com/help/secrets-manager-overview/
I can't actually find it on GitHub, but perhaps once it is out of beta.

**Reply by BlackDex** [reply]
`secrets-manager` is located here: https://github.com/bitwarden/clients/tree/master/bitwarden_license/bit-web/src/app
Which is bound to this https://github.com/bitwarden/clients/blob/master/LICENSE_BITWARDEN.txt license.
Which excludes it from being used/distributed as an OpenSourceSoftware (OSS) component.

**Reply by LucEast** [reply]
Do I get it right, that the Secrets-Manager is now open sourced? 
- https://www.reddit.com/r/Bitwarden/comments/15ya0ka/bitwarden_secrets_manager_is_now_generally/ 
- https://bitwarden.com/blog/secrets-manager-general-availability/

**Reply by BlackDex** [reply]
The source was already available if you would have checked my link.
Also checkout https://github.com/bitwarden/server/tree/master/bitwarden_license/src/Commercial.Core

It's all bound to the commercial license, and thus, we are not allowed to ship that code in any way, or we need to build it from scratch our self.

I think previously it wasn't available for everybody, but now it is.

**Reply by LucEast** [reply]
> I think previously it wasn't available for everybody, but now it is.
> 

Sorry for getting this wrong. I don't contribute that much and have thought that the fact that the Secrets Manager has been made FOSS automaticly allows to be used/contributed to.

**Reply by stefan0xC** [reply]
@LucEast Having the source code available does not make something FOSS. The Bitwarden project itself calls the [Bitwarden License a _"source available" license_](https://github.com/bitwarden/server/blob/master/LICENSE_FAQ.md#bitwarden-software-licensing). Because the commercial modules under the `bitwarden_license` folder require "a paid subscription" to use in production (among other restrictions), they are neither ["open source"](https://opensource.org/osd/) nor ["free software"](https://www.gnu.org/philosophy/free-sw.html).

**Reply by luckydonald** [reply]
![image](https://github.com/dani-garcia/vaultwarden/assets/2737108/80eb0262-db80-4e19-92b6-b762adf0bcaf)
so you can selfhost their product for $ 12 per month now.

**Comment by Dreamtrailtv** [comment]
is it possible that parts of the Secretsmanager sourcecode are available under the AGPL license in Github? https://github.com/bitwarden/server/tree/master/src/Core/SecretsManager

**Reply by BlackDex** [reply]
Without a GUI it is almost not usable.

**Reply by minionflo** [reply]
I think a cli is good enough for the secrets manager

**Comment by luckydonald** [comment]
Ho does it differ to the normal bitwarden?

I mean there always was the bitwarden CLI, and this feels super similar, except that it does that from within your own application process, not prior to launching anything.
That means - if I understood correctly - no leaking of the passwords via environment variables which could be seen by inspecting the running process.

So, that said, maybe this could be integrated into vaultwarden? Like just marking an password as available to a "service account"?

**Reply by BlackDex** [reply]
If you want something similar, use the Bitwarden cli with the `serve` command. Vaultwarden already supports the api tokens for users and organizations, so you can use that to login.

I don't that anything similar to secrets manager will be coming to Vaultwarden.

**Comment by tukusejssirs** [comment]
If it is source available only, why not create an open source way? Using a plugin if necessary.

I thing VV could provide a way to check passwords are leaked on the web and a way to use in CI (GitHub CI, GitLab CI, …). That would awesome! :heart:

**Reply by BlackDex** [reply]
Vaultwarden already support leaked password checks via the web-vault.

Adding custom features to the web-vault is not easy, and hard to maintain. So that will probably not happen.

If you want that feature, use the Bitwarden hosted vault, it has free support for this.

**Comment by minionflo** [comment]
Are there any updates on this?

**Reply by stefan0xC** [reply]
Nope. Bitwarden has not released the Secrets Manager under the GPLv3, so https://github.com/dani-garcia/vaultwarden/discussions/3368#discussioncomment-5428525 still applies.

**Reply by OrionCodeDev** [reply]
But maybe that part with AGPL can be supported?
https://github.com/dani-garcia/vaultwarden/discussions/3368#discussioncomment-6872293
@stefan0xC

**Reply by BlackDex** [reply]
There is no part we can add. It's useless without a GUI.
You need to be able to manage users, rights, secrets etc...
That isn't possible via the cli as far as i know.

**Comment by antoniolago** [comment]
Hello everyone, I created an app that makes VW a K8s secret manager by leveraging <strike>bw-cli</strike> Vaultwarden API, if anyone interested: https://github.com/antoniolago/vaultwarden-kubernetes-secrets

**Reply by derekslenk** [reply]
> Hello everyone, I created an app that makes VW a K8s secret manager by leveraging bw-cli, if anyone interested in helping me test it, check it out: https://github.com/antoniolago/vaultwarden-kubernetes-secrets

Exactly what I was looking for!

**Comment by danielefranceschi** [comment]
Considering that Bitwarden's Secrets Manager APIs are known https://github.com/bitwarden/sdk-sm/blob/main/crates/fake-server/src/lib.rs IMHO this could be feasible.

**Reply by BlackDex** [reply]
It's a Bitwarden Licenses feature, not open source.
There is no OSS UI available for the secrets manager.

**Reply by danielefranceschi** [reply]
So why not implementing an _independent_ oss ui? Or an oss cli? Or just the backend APIs and wait for someone implement the UI? Or it's something related to [this comment of yours](https://github.com/dani-garcia/vaultwarden/discussions/3368#discussioncomment-9536913)?

(I'd do it myself but I'm not proficient in UI or Rust, and I don't trust AI code I can't review)

**Reply by stefan0xC** [reply]
Because that is not the goal of this project and it frankly does not make much sense to develop an independent UI and then use that specific API because it's probably easier to create and develop against an independent API that you have more control over instead of integrating it with a separate code base like Vaultwarden. (And without an UI why would we want to implement the API ourselves? I mean that's just unnecessary work for a separate product that no one can use or develop against.)

**Comment by Sagit-chu** [comment]
Just a quick thought: the Hermes Agent framework actually just added native support for Bitwarden Secrets Manager. Given how fast things are moving in the AI space, we'll probably see a lot more AI agents starting to use BWS to handle their API keys very soon. It would be awesome to see this supported in Vaultwarden!

https://hermes-agent.nousresearch.com/docs/user-guide/secrets/bitwarden

**Reply by BlackDex** [reply]
Not gonna happen, read the comments above.
